libsolutil/CommonData.h

```diff
@@ -141,6 +142,36 @@ inline std::multiset<T...>& operator-=(std::multiset<T...>& _a, C const& _b)
 namespace solidity::util
 {
 
+/// Functional map.
+/// Returns a container _oc applying @param _op to each element in @param _c.
+/// By default _oc is a vector.
+/// If another return type is desired, an empty contained of that type
+/// is given as @param _oc.
+template<class Container, class Callable, class OutputContainer =
+	std::vector<std::invoke_result_t<
+		Callable,
+		decltype(*std::begin(std::declval<Container>()))
+>>>
+auto fmap(Container const& _c, Callable&& _op, OutputContainer _oc = OutputContainer{})
+{
+	std::transform(std::begin(_c), std::end(_c), std::back_inserter(_oc), _op);
```

Comment:
I wonder whether ``std::inserter(_oc, std::end(_oc))`` would be more general... that would work e.g. for ``std::set`` as well... on the other hand ``std::back_inserter`` is faster for vectors... the "proper" way to do this would be to use ``back_inserter``, if available, and fall back to ``inserter`` otherwise (e.g. using some ``custom_inserter`` that decides with SFINAE based on a specialization with ``std::void_t<decltype(std::back_inserter(_arg))>``)... such a thing was actually once proposed for the standard (https://isocpp.org/files/papers/p0471r0.html)... but not sure it's worth the effort...

Reply:
Since we can expect the speed-disadvantage of ``std::inserter(_oc, std::end(_oc))`` to be negligible for vectors (maybe even optimized away), I'd say just using that should be fine, if we don't want to bother with a custom one that distinguishes between cases.